### ISSUE TITLE
Authors column should be populated even if not uploaded by user

### ISSUE THREAD

**Issue by emanuil-tolev** [CottageLabs/LanternPM#53]
We've got articles where the compliance processing notes say "Added author list from EUPMC" and then there's no Author(s) column.

The reason for that is that users themselves can also upload Author(s) information, so if the column is not in the upload, we don't show it. See the if statement on https://github.com/CottageLabs/api/blob/7ee9e53f829324ffdf2e8090a0cb83e43070bea6/apps/clapi/server/endpoints/service/lantern.js#L198 (and note Author(s) in the definition of "columns to be treated this way" 5 lines above) .

My initial impulse is to just append it to the end of the `fields` array on https://github.com/CottageLabs/api/blob/7ee9e53f829324ffdf2e8090a0cb83e43070bea6/apps/clapi/server/endpoints/service/lantern.js#L186 . This seems to work.

Desired behaviour:
1. If an author list is retrievable from EUPMC or another source, overwrite contents of Author(s) field with the new data. Regardless of whether the user put anything in that field or not.
2. If an author list is not retrievable from our sources and
   
   a. if the user has uploaded Author(s) information for this record => do not overwrite whatever the user has uploaded
   b. if the user has left Author(s) empty for this record or not supplied the column at all => leave it blank

Additionally, if we find no Author(s) information for any article in the document, and the user has not uploaded the column, then we should still display it as part of the output - obviously all records will have this field blank in that case. In other words, Author(s) should function like the PMCID, PMID, DOI columns in this aspect.

I'll have a PR for this in a second. The change I propose above works fine, but there's also a bug in the formatwellcome function that breaks requirement 2a (we overwrite user content with EUPMC blanks unintentionally, because empty arrays in Javascript are still truthy).

**Comment by markmacgillivray** [comment]
This is actually a change to requirements then, not a bug fix. At the moment we do not output fields that the user does not put in their input list - you are now changing that. Also, you are saying that if EPMC returns no authors that we should assume the user-provided authors list to override that - changing the priority of what list is considered the correct information. The way you are proposing to do this would work, but will also change how we do things for Lantern. So @richard-jones will also have to consider changing the docs for both to indicate that Author(s) is a special field, treated differently from the others, if we go ahead with this change.

**Comment by emanuil-tolev** [comment]
> This is actually a change to requirements then, not a bug fix.

This was in the original requirements (output the column, also allow users to upload it). What's the point of fetching author data if we never display it? This isn't a field like University where we just regurgitate what the user gave us back to them, we actually have some new data to show. Shouldn't we display the column if so?

And all other columns of a similar nature (e.g. PMCID) work this way already. If I put in a PMCID and the system does not find a different one itself, it will give me my PMCID back. Try putting in PMC10000 together with some DOI that doesn't have its own PMCID.

**Comment by markmacgillivray** [comment]
I am confused. If we already do overwrite user-supplied info with what we do find, but do not overwrite user-supplied info when we do not find anything, then what are you saying the change is?

If an EUPMC record has no PMCID, then we are assuming that the record has no PMCID. If the user has provided a PMCID anyway, we give them back their PMCID. But if the EUPMC record has no authors, then the author list for that article is a list of nobody. So if the user has provided a different author list, we overwrite it with the list we have found (even if it is empty, it is still the author list we have found).

If it is supposed to output the authors list but does not, I understand that fix, and yes that would be to do with the truth check, that bit makes sense - but it still seems that you are suggesting some sort of change to how we handle author information, so surely at least our docs are ambiguous on this point?

**Comment by emanuil-tolev** [comment]
> If it is supposed to output the authors list but does not

That's basically it.

> it still seems that you are suggesting some sort of change to how we handle author information

Hmm.. do I? I don't mean to suggest that - sorry, perhaps the OP is too wordy and gives that impression. All this started from "the Author(s) column should be appearing, why isn't it" and then went to "how do I make it appear and make sure it works like all the other columns we fill with data, like PMCID, DOI etc.".

I found it a bit difficult to trace what the problem was, so wrote this issue with "full context from scratch" as it were.

**Comment by markmacgillivray** [comment]
OK, cool. As already agreed, the proposed change would work anyway - I was just wanting to make sure we think of docs and user management too, whenever considering changes that may alter functionality, so that Richard can be happy he has everything up to date and can keep users happy. This one sounded confusing, but if you're happy with it and we meet what we say we do, then go ahead with the change :)

**Comment by emanuil-tolev** [comment]
You are also correct, the docs don't mention it as compliance output, which it is (we gather new data). I will confirm with Richard and push an update to the docs if needed. Let's hold off merging this change until then.

**Comment by emanuil-tolev** [comment]
Confirmed it's ok to make the change, leaving open for me to amend the docs later today.

**Comment by emanuil-tolev** [comment]
Updated docs.

**Comment by emanuil-tolev** [comment]
Hmm, https://lantern.cottagelabs.com/docs not showing its new version, but https://wellcome.test.cottagelabs.com/docs _is_ showing its own updated file. 

https://lantern.cottagelabs.com/docs is serving this file https://github.com/CottageLabs/cl/blob/master/sub/lantern/content/docs.html on `master` branch, right? My new addition is both in develop and master versions of that file anyway.

EDIT: addition is https://github.com/CottageLabs/cl/blob/master/sub/lantern/content/docs.html#L221 (lines 220-223 in total).

**Comment by richard-jones** [comment]
done
